### PR TITLE
Prepare overall project structure

### LINKED ISSUE #10: Agree for project structure
# Project structure
- src
  - components
  - images
  - pages
  - styled-components
  - templates

### PR DESCRIPTION
**Pull request by tomaszgil**
Resolves #10

### CODE REVIEW

File: src/atoms/Container.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const Container = styled.div`
+  padding: 0 2rem;
+
+  ${({ fullWidth }) =>
+    fullWidth ? "width: 100%;" : "max-width: 1200px; margin: 0 auto;"}
+
+  ${({ theme }) => theme.media.mobile`
+    padding-left: 1rem;
+    padding-right: 1rem;
+  `};
```
Comment (rrudol) [inline src/atoms/Container.js]: how about tablet view? 🤔

File: src/components/HeroCarousel.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+import Container from "../atoms/Container"
+
+const HeroCarousel = ({ staticContent, dynamicList }) => (
+  <Container>
+    {staticContent}
```
Comment (rrudol) [inline src/components/HeroCarousel.js]: maybe `children` instead of `staticContent `?

File: src/components/MenuButton.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  outline: 0;
+  right: 0;
+  padding: 10px;
+  width: 40px;
+  position: relative;
+  z-index: 300;
+
+  &:focus,
+  &:active {
+    outline: 0;
```
Comment (rrudol) [inline src/components/MenuButton.js]: accessibility issue

File: src/components/MenuButton.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import PropTypes from "prop-types"
+
+const Button = styled.button`
+  display: none;
+  background: none;
+  border: 0;
+  cursor: pointer;
+  height: 40px;
+  margin-left: auto;
+  outline: 0;
```
Comment (rrudol) [inline src/components/MenuButton.js]: accessibility issue

File: src/components/MenuButton.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  &:active {
+    outline: 0;
+  }
+
+  ${({ theme }) => theme.media.tablet`
+    display: flex;
+  `};
+`;
+
+const Inner = styled.span`
```
Comment (rrudol) [inline src/components/MenuButton.js]: I'd call it HamburgerMenuIcon

File: src/containers/Header/Header.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import Logo from "../../atoms/Logo"
+import Container from "../../atoms/Container"
+import Navigation from "../Navigation"
+
+import { Wrapper, Content } from "./styled.js"
+
+function Header() {
+  return (
+    <Wrapper>
```
Comment (rrudol) [inline src/containers/Header/Header.js]: It should be named `HeaderWrapper`

File: src/containers/Header/Header.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      <Container>
+        <Content>
```
Comment (rrudol) [inline src/containers/Header/Header.js]: `HeaderContent`

File: src/containers/Header/styled.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import styled from "styled-components";
+
+export const Wrapper = styled.header`
+  padding: 1rem 0;
+`;
+
+export const Content = styled.div`
```
Comment (rrudol) [inline src/containers/Header/styled.js]: `HeaderContent`

File: src/containers/Header/styled.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import styled from "styled-components";
+
+export const Wrapper = styled.header`
```
Comment (rrudol) [inline src/containers/Header/styled.js]: `HeaderWrapper`

File: src/containers/Layout/Layout.js [hunk 10]
```diff
@@ -18,11 +20,14 @@ const Layout = ({ children }) => (
...
-        <Header siteTitle={data.site.siteMetadata.title} />
-        {children}
-        <Footer />
-      </>
+      <ThemeProvider theme={theme}>
+        <>
+          <Header siteTitle={data.site.siteMetadata.title} />
+          {children}
+          <Footer />
+          <GlobalStyle />
```
Comment (rrudol) [inline src/containers/Layout/Layout.js]: GlobalStyle on the bottom?

File: src/containers/Navigation/Navigation.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  const [menuState, setMenuState] = useState(false)
+
+  const onLinkClick = () => setMenuState(false)
+  const onOverlayClick = () => setMenuState(false)
+  const onMenuClick = () => setMenuState(!menuState)
+
+  return (
+    <>
+      <Nav isMenuOpened={menuState}>
+        <Wrapper isMenuOpened={menuState}>
```
Comment (rrudol) [inline src/containers/Navigation/Navigation.js]: `NavWrapper`

File: src/containers/Navigation/Navigation.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+          <List>
```
Comment (rrudol) [inline src/containers/Navigation/Navigation.js]: NavList

File: src/containers/Navigation/Navigation.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+            <li>
+              <MainNavLink as={Link} to="/submit" onClick={onLinkClick}>
+                Submit
+              </MainNavLink>
+            </li>
+          </List>
+        </Wrapper>
+      </Nav>
+      <MenuButton onClick={onMenuClick} isMenuOpened={menuState} />
+      <Overlay isVisible={menuState} onClick={onOverlayClick} />
```
Comment (rrudol) [inline src/containers/Navigation/Navigation.js]: NavOverlay

File: src/containers/Navigation/styled.js [hunk 14]
```diff
@@ -0,0 +1,89 @@
+import styled, { css } from "styled-components"
+
+import NavLink from "../../atoms/NavLink";
+
+export const Overlay = styled.div`
```
Comment (rrudol) [inline src/containers/Navigation/styled.js]: NavOverlay

File: src/containers/Navigation/styled.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+    padding-left: 1rem;
+    padding-right: 1rem;
+  `};
+`
+
+export const MainNavLink = styled(NavLink)`
+  margin-right: 3rem;
+`;
+
+export const Wrapper = styled.div`
```
Comment (rrudol) [inline src/containers/Navigation/styled.js]: NavWrapper

File: src/containers/Navigation/styled.js [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+    ${({ isMenuOpened }) => isMenuOpened &&
+      `
+      opacity: 1;
+      transition-delay: 0.2s;
+    `}
+  `};
+`;
+
+export const List = styled.ul`
```
Comment (rrudol) [inline src/containers/Navigation/styled.js]: NavList

File: src/containers/Navigation/styled.js [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+  margin: 0;
+  display: flex;
+  align-items: center;
+
+  ${({ theme }) => theme.media.tablet`
+    display: block;
+    margin-bottom: 2rem;
+  `};
+`;
+
```
Comment (rrudol) [inline src/containers/Navigation/styled.js]: Would be great if the order of appearing styles would match to order in Navigation.js

File: src/components/MenuButton.js [hunk 18]
```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import styled from "styled-components"
+import PropTypes from "prop-types"
+
+const Button = styled.button`
```
Comment (rrudol) [inline src/components/MenuButton.js]: We already have component called `Button`

File: src/atoms/Button.js [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+  max-width: 12em;
+  min-width: 6em;
+  padding: 0.75rem 2rem;
+  transition: 
+    background ${({ theme }) => theme.effects.transition.quick};
+  text-align: center;
+  text-decoration: none;
+
+  &:hover {
+    background: ${({ theme }) => theme.color.accent.primary};
```
Comment (rrudol) [inline src/atoms/Button.js]: text color should change too because of lack of contrast

File: src/atoms/Container.js (hunk 1 above)
Comment (tomaszgil) [reply src/atoms/Container.js]: Edge to edge with 2rem of padding seems fine, what would you like to have there?

File: src/components/HeroCarousel.js (hunk 2 above)
Comment (tomaszgil) [reply src/components/HeroCarousel.js]: I thought about it as well, that's good as well.

File: src/atoms/Container.js (hunk 1 above)
Comment (rrudol) [reply src/atoms/Container.js]: agh, I miss that `2rem` 😶 so it's okey 🙂

**Comment by tomaszgil** [comment]
Containers for me are react components that are predominantly responsible for logic (not only _containing_ presentational components). So SEO and other fit also here. If you don't like it, let's come up with a different name or move SEO component to, lets say, _helpers_ folder.

I don't like the idea of having a separate `styled-components` directory, because we also use styled components inside `containers` folder. And the name indicates use of the library which is not necessary - the clue here is that these are presentational and reusable components.

I don't see the need of having a separate folder for each presentational component given that each folder will contain one and only one file.

**Comment by tomaszgil** [comment]
Currently, svg files cannot be served with gatsby for inline use.
https://github.com/gatsbyjs/gatsby/issues/10297

**Pull request merged**